encodings/alp/src/alp_rd/serde.rs

```diff
@@ -14,12 +13,17 @@ use vortex_error::{VortexError, VortexExpect, VortexResult, vortex_bail, vortex_
 use super::{ALPRDEncoding, RDEncoder};
 use crate::ALPRDArray;
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Clone, prost::Message)]
 pub struct ALPRDMetadata {
-    right_bit_width: u8,
-    dict_len: u8,
-    dict: [u16; 8],
-    left_parts_ptype: PType,
+    #[prost(uint32, tag = "1")]
+    right_bit_width: u32,
+    #[prost(uint32, tag = "2")]
+    dict_len: u32,
+    #[prost(uint32, repeated, tag = "3")]
+    dict: Vec<u32>,
+    #[prost(enumeration = "PType", tag = "4")]
+    left_parts_ptype: i32,
```

Comment:
blergh, is there a way to make this PType in code even if its serialized as i32?

Reply:
there's not, you get a top level method that gives you back PType so you never have to touch this

Reply:
I'm not sure this is true, Onur just did this for DictLayout metadata